test/infrastructure/inmemory/internal/server/api/handler.go

```diff
@@ -488,7 +500,11 @@ func (h *apiServerHandler) apiV1Patch(req *restful.Request, resp *restful.Respon
 		return
 	}
 	if err := cloudClient.Patch(ctx, obj, patch); err != nil {
-		_ = resp.WriteErrorString(http.StatusInternalServerError, err.Error())
+		if status, ok := err.(apierrors.APIStatus); ok || errors.As(err, &status) {
```

Comment:
Should we do the same in l.499?

Reply:
I don't think so, the fact that we should read from cache before patching is an internal implementation detail, and if this operation fails I'm ok in surfacing this as an internal error.
Instead, we should return as API errors eventual issues that happens when calling patch